src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -261,3 +261,67 @@ function chol_blocked_rev(Σ̄::AbstractMatrix, L::AbstractMatrix, nb::Integer,
     # Convert to `Matrix`s because blas functions require StridedMatrix input.
     return chol_blocked_rev!(Matrix(Σ̄), Matrix(L), nb, upper)
 end
+
+#####
+##### `qr`
+#####
+
+const QR_TYPE = Union{QR, QRPivoted, LinearAlgebra.QRCompactWY}
+
+function rrule(::typeof(qr), X::AbstractMatrix{<:Real})
+    F = qr(X)
+
+    if size(F.R, 2) != size(F.R, 1)
+        throw(ArgumentError("Pullback for QR decomposition is only supported for m × n matrices with m >= n"))
+    end
```

Comment:
Why?

Reply:
Because I have not found a reference for that case.

Reply:
Section 3.2 of https://arxiv.org/pdf/2009.10071.pdf gives an `rrule` for `qr` for wide matrices (`m<n`). I haven't tested it, though.

Reply:
I will look into adding that.